middle_end/flambda2/simplify/simplify_common.ml

```diff
@@ -133,6 +141,7 @@ let split_direct_over_application apply ~param_arity ~result_arity
       ~inlining_state:(Apply.inlining_state apply)
       ~probe_name:(Apply.probe_name apply) ~position:(Apply.position apply)
       ~relative_history:(Apply.relative_history apply)
+      ~region:current_region
   in
   let perform_over_application_free_names =
     Apply.free_names perform_over_application
```

Comment:
The comment a few lines below (starting at line 153 on this PR) is misleading; I understand it as meaning that only the second application is between `Begin_region` and `End_region`. The code instead puts the region around both applications, which is what earlier comments imply, and which seems the correct behaviour to me.